Yehor/reactNetflix/src/pages/detailedPage/DetailedPage.tsx

```diff
@@ -1,90 +1,108 @@
 
-class DetailedPage extends React.Component<DetailedPageProps, {}> {
-  componentDidMount() {
-    const filmId = this.props.match.params.id;
-    console.log(filmId);
-    this.props.currentMovieLoad("movie is loading", filmId);
-    
-  }
+const DetailedPage = (props: DetailedPageProps) => {
+  useEffect(() => {
+    props.currentMovieLoad({
+      status: "movie is loading",
+      id: props.match.params.id,
+    });
+    return () => {
+      props.currentMovieReceived({ status: "creaned", movie: null });
+    };
+  }, [props.match.params.id]);
 
-  componentDidUpdate(prevProps: DetailedPageProps) {
-    if (this.props.match.params.id !== prevProps.match.params.id) {
-      this.props.currentMovieLoad("movie is loading", this.props.match.params.id);
-    }
-    if(this.props.movie !== prevProps.movie) {
-      const searchBy = "genres"
-      const search = this.props.movie.genres[0]
-      this.props.loadData("movies by genre is loading", "release_date", searchBy, search, this.props.offset)
+  useEffect(() => {
+    if (props.movie) {
+      const searchBy = "genres";
+      const search = props.movie.genres[0];
+      props.loadData({
+        sortBy: "release_date",
+        searchBy: searchBy,
+        search: search,
+        offset: props.offset,
+      });
     }
-  }
- 
+  }, [props.movie, props.offset]);
 
-  render() {
-    const { movie, movies } = this.props;
-    console.log(movie);
-
-    return (
-      <div className="allpageWrapper">
-        <div className="wrapper">
-          <div className="blur">
-            <div className="top-container">
-              <Header />
-              {movie ? (
-                <DetailedMovie movie={movie} />
-              ) : (
-                <div>There is no movie</div>
-              )}
-            </div>
+  const { movie, movies } = props;
+  return (
+    <div className="allpageWrapper">
+      <div className="wrapper">
+        <div className="blur">
+          <div className="top-container">
+            <Header />
+            {movie ? (
+              <DetailedMovie movie={movie} />
+            ) : (
+              <div>There is no movie</div>
+            )}
           </div>
         </div>
-        <div className="main-container">
-          {movie ? <SortByGenrePanel genre={movie.genres[0]}/>: <SortByGenrePanel genre={"genre"}/>}
-        </div>
-        <div className="main-container">
+      </div>
+      <div className="main-container">
+        {movie ? (
+          <SortByGenrePanel genre={movie.genres[0]} />
+        ) : (
+          <SortByGenrePanel genre={"genre"} />
+        )}
+      </div>
+      <div className="main-container">
         <MoviesList movies={movies} />
+        <div className="pagination">
+          <button onClick={props.dataOffsetDecrement}>Back</button>
+          <span className="pages">
+            {props.offset ? props.offset / 9 + 1 : 1}
+          </span>
+          <button onClick={props.dataOffsetIncrement}>Next</button>
```

Comment:
Movies are not endless, would be grate if you wrapper your button in some condition. Hide button if there is no more movie left to fetch